src/main/java/core/basesyntax/dao/DriverJdbcDaoImpl.java

```diff
@@ -0,0 +1,110 @@
+                driver.setId(result.getObject(1, Long.class));
+            }
+            return driver;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create driver " + driver, e);
+        }
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String query = "SELECT * FROM drivers WHERE driver_id = ? AND deleted = FALSE";
+        Driver driver = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement gettingDriver = connection.prepareStatement(query)) {
+            gettingDriver.setLong(1, id);
+            ResultSet result = gettingDriver.executeQuery();
+            if (result.next()) {
+                driver = getDriver(result);
+            }
+            return Optional.ofNullable(driver);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get driver by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String query = "SELECT * FROM drivers WHERE deleted = FALSE";
+        List<Driver> allDrivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement gettingAll = connection.prepareStatement(query)) {
+            ResultSet result = gettingAll.executeQuery();
+            while (result.next()) {
+                allDrivers.add(getDriver(result));
+            }
+            return allDrivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all drivers from database", e);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String query = "UPDATE drivers SET name = ?, licenseNumber = ? "
+                + " WHERE driver_id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updatedDriver = connection.prepareStatement(query)) {
+            updatedDriver.setString(1, driver.getName());
+            updatedDriver.setString(2, driver.getLicenseNumber());
+            updatedDriver.setObject(3, driver.getId());
+            updatedDriver.executeUpdate();
+            return driver;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update driver " + driver, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE drivers SET deleted = TRUE WHERE driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deletedDriver = connection.prepareStatement(query)) {
+            deletedDriver.setLong(1, id);
+            return deletedDriver.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t remove driver by id " + id, e);
+        }
+    }
+
+    private Driver getDriver(ResultSet result) {
```

Comment:
```suggestion
    static Driver getDriver(ResultSet result) {
```
it will make it package-private and you won't need to duplicate this code in `CarJdbcDaoImpl`. same with `getManufacturer` if u have it ManufacturerDao implementation